### DISCUSSION TITLE
Meaning of annotation RequiresApi for block statement

### DISCUSSION THREAD

**Discussion by KvanTTT** [shadowsocks/shadowsocks-android#2884] [General]
Hi! I'm Kotlin compiler developer and now I'm working on the new version of compiler (K2). One part of my work is testing user projects. I found your project and encountered the failing on the following code (`@RequiresApi` annotation is not applicable to block statement):

https://github.com/shadowsocks/shadowsocks-android/blob/3d09ab407183b3ed7f260233966a780d5584a6fb/core/src/main/java/com/github/shadowsocks/Core.kt#L155-L166

Could you elaborate what did you mean by placing the annotation to the block? It looks like it has no effect and it's a bug in the old version of compiler (K1) since annotation for block is not allowed even by grammar: https://kotlinlang.org/docs/reference/grammar.html#block

**Comment by Mygod** [comment]
This is for muting an error in Android Studio. It seems IntelliJ parses Kotlin differently from the compiler.

**Reply by KvanTTT** [reply]
Could you describe this error in Anroid Studio?

**Reply by Mygod** [reply]
Honestly I don't remember. Now that I look at it, this is probably unnecessary.
